runtime/resolver.go

```diff
@@ -290,17 +302,19 @@ func (r *driverResolverResult) MarshalJSON() ([]byte, error) {
 		}
 		row, ok := ret.(map[string]any)
 		if !ok {
-			// this should never happen
 			return nil, fmt.Errorf("Resolver.MarshalJSON unexpected type %T", ret)
 		}
 		out = append(out, row)
 	}
 	if r.rows.Err() != nil {
 		return nil, r.rows.Err()
 	}
-	if out == nil { // fixes 'null' output when there are no rows
+	if out == nil {
 		out = []map[string]any{}
 	}
+	if r.meta != nil {
+		return json.Marshal(map[string]any{"data": out, "meta": r.meta})
+	}
 	return json.Marshal(out)
```

Comment:
`MarshalJSON` is supposed to always return a JSON array (equivalent to paging through rows individually with `Next` and then serializing with JSON).

You should add a new `Meta` function on the `ResolverResult` interface for returning it instead.